disputer/disputer.js

```diff
@@ -128,23 +128,26 @@ class Disputer {
       const dispute = this.empContract.methods.dispute(disputeableLiquidation.id, disputeableLiquidation.sponsor);
 
       // Simple version of inventory management: simulate the transaction and assume that if it fails, the caller didn't have enough collateral.
+      let gasLimit;
       try {
-        await dispute.call({ from: this.account, gasPrice: this.gasEstimator.getCurrentFastPrice() });
+        await dispute.call({ from: this.account });
+        gasLimit = Math.floor((await dispute.estimateGas({ from: this.account })) * 1.5);
       } catch (error) {
         this.logger.error({
           at: "Disputer",
           message: "Cannot dispute liquidation: not enough collateral (or large enough approval) to initiate dispute✋",
           disputer: this.account,
           sponsor: disputeableLiquidation.sponsor,
           liquidation: disputeableLiquidation,
+          totalPaid,
           error
         });
         continue;
       }
 
       const txnConfig = {
         from: this.account,
-        gas: this.txnGasLimit,
+        gas: gasLimit <= this.txnGasLimit ? gasLimit : this.txnGasLimit,
```

Comment:
Maybe just simplify as `Math.min(gasLimit, this.txnGasLimit)`

Reply:
sure & done.